### ISSUE TITLE
failed to rename container

### ISSUE THREAD

**Issue by charleshan** [dokku/dokku#3900]
```
Error response from daemon: sandbox 96264d25ae593034f4c34de6b1f24e71e0f4ce76bd9bcef413e612ef440f8f76 not found
Error: failed to rename container named zealous_cori
```

Dokku fails to rename the containers during deployment/build. I didn't think much of it because dokku cleanup usually fixed the issue but this resulted in having multiple containers that were running simultaneously which resulted in maxing out all the resources.

I'm using Ubuntu 18.04.3 and dokku v0.19.13

<details><summary>dokku report</summary>

```
-----> uname: Linux 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                     total        used        free      shared  buff/cache   available
       Mem:           7976         859        3821           0        3295        6868
       Swap:         16383           0       16383
-----> docker version:
       Client: Docker Engine - Community
        Version:           19.03.6
        API version:       1.40
        Go version:        go1.12.16
        Git commit:        369ce74a3c
        Built:             Thu Feb 13 01:27:49 2020
        OS/Arch:           linux/amd64
        Experimental:      false

       Server: Docker Engine - Community
        Engine:
         Version:          19.03.6
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.12.16
         Git commit:       369ce74a3c
         Built:            Thu Feb 13 01:26:21 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.2.10
         GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
        runc:
         Version:          1.0.0-rc8+dev
         GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info:
WARNING: No swap limit support
       Client:
        Debug Mode: true

       Server:
        Containers: 13
         Running: 2
         Paused: 0
         Stopped: 11
        Images: 9
        Server Version: 19.03.6
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
        runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.15.0-91-generic
        Operating System: Ubuntu 18.04.3 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 4
        Total Memory: 7.789GiB
        Name: project-abc
        ID: NLJQ:OOPW:OSBN:YA5T:DYOW:EJQ6:KZCV:ARYR:BISH:6GXX:5KJC:OMSE
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> sigil version: 0.4.0
-----> herokuish version:
       herokuish: 0.5.6
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v207
         heroku-buildpack-nodejs    v167
         heroku-buildpack-clojure   v84
         heroku-buildpack-python    v163
         heroku-buildpack-java      v66
         heroku-buildpack-gradle    v31
         heroku-buildpack-scala     v87
         heroku-buildpack-play      v26
         heroku-buildpack-php       v169
         heroku-buildpack-go        v136
         buildpack-nginx            v12
-----> dokku version: dokku version 0.19.13
-----> dokku plugins:
       plugn: 0.3.2
         00_dokku-standard    0.19.13 enabled    dokku core standard plugin
         20_events            0.19.13 enabled    dokku core events logging plugin
         app-json             0.19.13 enabled    dokku core app-json plugin
         apps                 0.19.13 enabled    dokku core apps plugin
         builder-dockerfile   0.19.13 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.19.13 enabled    dokku core builder-herokuish plugin
         buildpacks           0.19.13 enabled    dokku core buildpacks plugin
         certs                0.19.13 enabled    dokku core certificate management plugin
         checks               0.19.13 enabled    dokku core checks plugin
         common               0.19.13 enabled    dokku core common plugin
         config               0.19.13 enabled    dokku core config plugin
         docker-options       0.19.13 enabled    dokku core docker-options plugin
         domains              0.19.13 enabled    dokku core domains plugin
         enter                0.19.13 enabled    dokku core enter plugin
         git                  0.19.13 enabled    dokku core git plugin
         letsencrypt          0.9.2 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.19.13 enabled    dokku core logs plugin
         network              0.19.13 enabled    dokku core network plugin
         nginx-vhosts         0.19.13 enabled    dokku core nginx-vhosts plugin
         plugin               0.19.13 enabled    dokku core plugin plugin
         proxy                0.19.13 enabled    dokku core proxy plugin
         ps                   0.19.13 enabled    dokku core ps plugin
         repo                 0.19.13 enabled    dokku core repo plugin
         resource             0.19.13 enabled    dokku core resource plugin
         scheduler-docker-local 0.19.13 enabled    dokku core scheduler-docker-local plugin
         shell                0.19.13 enabled    dokku core shell plugin
         ssh-keys             0.19.13 enabled    dokku core ssh-keys plugin
         storage              0.19.13 enabled    dokku core storage plugin
         tags                 0.19.13 enabled    dokku core tags plugin
         tar                  0.19.13 enabled    dokku core tar plugin
         trace                0.19.13 enabled    dokku core trace plugin
```

</details>

**Comment by josegonzalez** [comment]
How can I replicate this? Seems pretty bad, but we should definitely be keeping track of the container...

**Comment by charleshan** [comment]
I'm not sure. I updated herokuish to 0.5.7 just now but I'm just taking a shot in the dark. This didn't work either...

Here's some info that might be interesting.

```
$ dokku deploy earth
...
-----> Creating https nginx.conf
       Reloading nginx
-----> Renaming containers
       Found previous container(s) (d1f111911903) named earth.beat.1
       Renaming container (d1f111911903) earth.beat.1 to earth.beat.1.1584590718
       Renaming container (1e10d440fe05) peaceful_shamir to earth.beat.1
       Found previous container(s) (b025b077639b) named earth.web.1
       Renaming container (b025b077639b) earth.web.1 to earth.web.1.1584590718
       Renaming container (9e0567c26b05) admiring_haslett to earth.web.1
       Renaming container (44207b4c3b4a) unruffled_curran to earth.worker.1
Error response from daemon: sandbox adaaf953c602f59ec305171d04ce041b3ea9f1955c3807b44c22c4022950caa9 not found
Error: failed to rename container named unruffled_curran

$ docker ps
CONTAINER ID        IMAGE                COMMAND             CREATED              STATUS                          PORTS               NAMES
1e10d440fe05        dokku/earth:latest   "/start beat"       About a minute ago   Up About a minute                                   earth.beat.1
0b1e7f7abaa7        dokku/earth:latest   "/start worker"     About a minute ago   Restarting (1) 9 seconds ago                        silly_euler
c7a52229dd1c        dokku/earth:latest   "/start worker"     About a minute ago   Restarting (1) 12 seconds ago                       boring_bartik
44207b4c3b4a        dokku/earth:latest   "/start worker"     About a minute ago   Restarting (1) 13 seconds ago                       unruffled_curran
9e0567c26b05        dokku/earth:latest   "/start web"        About a minute ago   Up About a minute                                   earth.web.1
d1f111911903        564b4890228a         "/start beat"       5 minutes ago        Up 5 minutes                                        earth.beat.1.1584590718
b025b077639b        564b4890228a         "/start web"        5 minutes ago        Up 5 minutes                                        earth.web.1.1584590718
8291f320f674        52f811ccefc1         "/start beat"       15 minutes ago       Up 15 minutes                                       earth.beat.1.1584590482
49890f98cb7c        52f811ccefc1         "/start web"        15 minutes ago       Up 15 minutes                                       earth.web.1.1584590482

$ dokku ps:report earth
=====> earth ps information
       Processes:                     5
       Deployed:                      true
       Running:                       true
       Restore:                       true
       Restart policy:                on-failure:10
       Ps can scale:                  true
       Status beat.1:                 running    (CID: 1e10d440fe05)
       Status web.1:                  running    (CID: 9e0567c26b05)
       Status worker.1:               restarting (CID: 44207b4c3b4a)
       Status worker.2:               restarting (CID: c7a52229dd1c)
       Status worker.3:               restarting (CID: 0b1e7f7abaa7)
```

After a few minutes:
```
$ docker ps
CONTAINER ID        IMAGE                COMMAND             CREATED             STATUS              PORTS               NAMES
1e10d440fe05        dokku/earth:latest   "/start beat"       2 minutes ago       Up 2 minutes                            earth.beat.1
9e0567c26b05        dokku/earth:latest   "/start web"        2 minutes ago       Up 2 minutes                            earth.web.1
d1f111911903        564b4890228a         "/start beat"       6 minutes ago       Up 6 minutes                            earth.beat.1.1584590718
b025b077639b        564b4890228a         "/start web"        6 minutes ago       Up 6 minutes                            earth.web.1.1584590718
8291f320f674        52f811ccefc1         "/start beat"       16 minutes ago      Up 16 minutes                           earth.beat.1.1584590482
49890f98cb7c        52f811ccefc1         "/start web"        16 minutes ago      Up 16 minutes                           earth.web.1.1584590482

$ dokku ps:report earth
=====> earth ps information
       Processes:                     5
       Deployed:                      true
       Running:                       mixed
       Restore:                       true
       Restart policy:                on-failure:10
       Ps can scale:                  true
       Status beat.1:                 running    (CID: 1e10d440fe05)
       Status web.1:                  running    (CID: 9e0567c26b05)
       Status worker.1:               exited     (CID: 44207b4c3b4a)
       Status worker.2:               exited     (CID: c7a52229dd1c)
       Status worker.3:               exited     (CID: 0b1e7f7abaa7)
```

This tells me that:
1. The worker containers are failing to stay alive (could be a separate issue).
2. Other containers are duplicated.

**Comment by josegonzalez** [comment]
Is your app actually starting correctly? Seems like it is restarting...

Also yeah if we fail to restart because the container isn't running, that should def be placed on the kill list....

**Comment by charleshan** [comment]
> Is your app actually starting correctly? Seems like it is restarting...
`web` and `beat` containers start correctly. The restarting containers are the worker containers which are removed a few minutes later.

**Comment by charleshan** [comment]
"Error: Failed to rename container" doesn't always happen. I just ran `dokku deploy earth` twice in a row. The first time it deployed without the rename error (the worker containers still died). The second time, there was the rename error.

🤯

```
$ dokku deploy earth
       Default container check successful!
-----> Running post-deploy
-----> Configuring ketchup.today...(using built-in template)
-----> Creating https nginx.conf
       Reloading nginx
-----> Renaming containers
       Found previous container(s) (5eb27bd94d17) named earth.beat.1
       Renaming container (5eb27bd94d17) earth.beat.1 to earth.beat.1.1584594994
       Renaming container (8fd64a4f16f0) elastic_tesla to earth.beat.1
       Found previous container(s) (ab15aab5ef6c) named earth.web.1
       Renaming container (ab15aab5ef6c) earth.web.1 to earth.web.1.1584594994
       Renaming container (eff6fb7f51c6) heuristic_albattani to earth.web.1
       Renaming container (cf38e608108a) cranky_boyd to earth.worker.1
       Renaming container (a6f6cfb4df94) busy_northcutt to earth.worker.2
       Renaming container (f7bfec94e925) pedantic_swanson to earth.worker.3
-----> Shutting down old containers in 60 seconds
       5eb27bd94d17607a7f49bdd0314c2d47fdadc4ecdda0a30dbf0b0b711d5670fd
       ab15aab5ef6ce3a0e8f840322cbde5aeb44b0066b267609b62b8e44c4fb54d10
       625eb513b31a6f71f824b09c8f2811fc61a59e1a28c4170cb718b1545d74d6ff
       232777a3826ce512fdb881c1b7940eb8d12c4379606c0736d608edff706e08bd
       80b8e562ef2995fb0e756c4be337bff716eade7adadc1b5e79b7dad1ac5383fa
```

**Comment by josegonzalez** [comment]
I'm thinking it just so happens that the containers are running when the rename happened, so its something of a race condition. We should definitely handle that case, though I'm not sure if we should call that deploy failed or what...

Can you redo that failed deploy and call the following on the containers that fail to be renamed?

```shell
# replace $CID with the container id
docker inspect $CID
docker logs $CID
```

**Comment by charleshan** [comment]
Sure, here you go.

```
       Default container check successful!
-----> Running post-deploy
-----> Configuring project.earth...(using built-in template)
-----> Creating https nginx.conf
       Reloading nginx
-----> Renaming containers
       Found previous container(s) (8c77fcd78696) named earth.beat.1
       Renaming container (8c77fcd78696) earth.beat.1 to earth.beat.1.1584636414
       Renaming container (1c9473a2a082) agitated_curran to earth.beat.1
       Found previous container(s) (cf47af94d8c3) named earth.release.1
       Renaming container (cf47af94d8c3) earth.release.1 to earth.release.1.1584636414
       Renaming container (b406543bea02) reverent_wing to earth.release.1
       Found previous container(s) (3b1bf0592140) named earth.web.1
       Renaming container (3b1bf0592140) earth.web.1 to earth.web.1.1584636415
       Renaming container (01421a6dca72) sleepy_chaum to earth.web.1
       Found previous container(s) (8cbe95f15657) named earth.worker.1
       Renaming container (8cbe95f15657) earth.worker.1 to earth.worker.1.1584636415
       Renaming container (a2905ca0f71c) focused_varahamihira to earth.worker.1
Error response from daemon: sandbox fc4d50a7f5892e8d3b0282d01340e044305c5f1b42bf7fb7c99c2c5ae27174a9 not found
Error: failed to rename container named focused_varahamihira

$ docker ps
CONTAINER ID        IMAGE                COMMAND             CREATED             STATUS                  PORTS               NAMES
1c9473a2a082        dokku/earth:latest   "/start beat"       27 seconds ago      Up 26 seconds                               earth.beat.1
a2905ca0f71c        dokku/earth:latest   "/start worker"     30 seconds ago      Up Less than a second                       focused_varahamihira
01421a6dca72        dokku/earth:latest   "/start web"        35 seconds ago      Up 34 seconds                               earth.web.1
8c77fcd78696        22d884ced803         "/start beat"       4 minutes ago       Up 4 minutes                                earth.beat.1.1584636414
3b1bf0592140        22d884ced803         "/start web"        4 minutes ago       Up 4 minutes                                earth.web.1.1584636415

$ docker inspect a2905ca0f71c
[
    {
        "Id": "a2905ca0f71cf0e071c7bdd402886465c63bd62af8cece5a39d4b9e9f5cff69a",
        "Created": "2020-03-19T16:46:34.85170591Z",
        "Path": "/start",
        "Args": [
            "worker"
        ],
        "State": {
            "Status": "restarting",
            "Running": true,
            "Paused": false,
            "Restarting": true,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2020-03-19T16:47:14.121363996Z",
            "FinishedAt": "2020-03-19T16:47:16.996350629Z"
        },
        "Image": "sha256:f419bee69a4c590833e1d8db4769285defe58080f5ea4ffa955078ac141c4429",
        "ResolvConfPath": "/var/lib/docker/containers/a2905ca0f71cf0e071c7bdd402886465c63bd62af8cece5a39d4b9e9f5cff69a/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/a2905ca0f71cf0e071c7bdd402886465c63bd62af8cece5a39d4b9e9f5cff69a/hostname",
        "HostsPath": "/var/lib/docker/containers/a2905ca0f71cf0e071c7bdd402886465c63bd62af8cece5a39d4b9e9f5cff69a/hosts",
        "LogPath": "/var/lib/docker/containers/a2905ca0f71cf0e071c7bdd402886465c63bd62af8cece5a39d4b9e9f5cff69a/a2905ca0f71cf0e071c7bdd402886465c63bd62af8cece5a39d4b9e9f5cff69a-json.log",
        "Name": "/focused_varahamihira",
        "RestartCount": 8,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "on-failure",
                "MaximumRetryCount": 10
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/2e102ab7bfff01ee50fd08f077bb9897076e93e9b4d84390e1a8d1ffc40dd497-init/diff:/var/lib/docker/overlay2/7971d3102708f00ac114daa7e913bf1e98af6335bb43b822b1c3feef0ed82e62/diff:/var/lib/docker/overlay2/e2b0ae190857429f12fcb18b54f897487725f62271f8e3c2a04ff66c73744720/diff:/var/lib/docker/overlay2/d57ab2000cb87b56d52eda727cb220dd3c93443707e72cf4099e2d0a1daf68f4/diff:/var/lib/docker/overlay2/b403151ecd3e33eae450a37d099ff5d482e012dcd034fe2a5ba683679c760f39/diff:/var/lib/docker/overlay2/07686c00819a988a5e87b7cf9c26c63a2309278078862a8629e2bf0b7fcd62a5/diff:/var/lib/docker/overlay2/ab2a39db7e9befac66acd879f4b0140775e812ebd029da84dfecc96d32978393/diff:/var/lib/docker/overlay2/d4a9616c1a777664c122315c3622f0098990180abbf1b918534700be8d42a49d/diff:/var/lib/docker/overlay2/5756f9de852566919f73424a1c28fa96100cad0fd9a1e45dc5980be8074634ad/diff:/var/lib/docker/overlay2/e2496ad07897516cf20f8182f1cde4840ca3cde19842d33c79a2021fb7a6ba53/diff:/var/lib/docker/overlay2/c22f0706138d40c64505c516b37c6bef8d188cbfd75574e6f82248bacb24653c/diff:/var/lib/docker/overlay2/994f9caa223f8220288f5f5a8a4ef957b8a77228924e78b2a7da789d82410d23/diff:/var/lib/docker/overlay2/986f90f7ae7576f2d582705d845c5128f4e51350a41b1506e49c941017828e17/diff:/var/lib/docker/overlay2/8553eb46a2bc5476ee629b745625953cb897377e78f61ba861798a70d98541f2/diff:/var/lib/docker/overlay2/4e6fa8248bf792709355855bf999ba80cd8282c42ad652821cacc288673ff1ca/diff:/var/lib/docker/overlay2/826a3f06cc3203da00a78813fda257c0e7e4df07114572c5caecd10629e9e238/diff:/var/lib/docker/overlay2/123b7bb31ca3d87355594fb2df5ed98025bb06b5b7aa302dc54cdc9e4fe9dd43/diff:/var/lib/docker/overlay2/fdbbe013b8c31a506b53c5207aa8f0bbab1ccdf193af713d1aa7d5fb0157976b/diff:/var/lib/docker/overlay2/f9b3fe94326ce279dd712286b1e495b296dc28f09822a10d186c69f843b29509/diff:/var/lib/docker/overlay2/0aea6954eb428e255eeb9f73607c86dacf0076efcb34d2966344a755176fcea7/diff:/var/lib/docker/overlay2/f2a79b0637b197a13a5aeb58a71b4aafc85666124c810a0a7a94823cb1940dcf/diff:/var/lib/docker/overlay2/29e9299cddeace7e13705d27cdd336b0ab70f7d7a511c41cacfc40d606ccc31e/diff:/var/lib/docker/overlay2/6ea7e8acc52ceb3ca4ff1825d569bc553eff147f3c1594ad54104bba8a46bbf5/diff:/var/lib/docker/overlay2/6274e961a856fff554d0b824cb88ce5b80a19ca41ac08c72d3ed960bab5ddc3c/diff:/var/lib/docker/overlay2/70d35753edd628de22252740ce1d5393fbf9936295130615fbefb0ad59312b30/diff",
                "MergedDir": "/var/lib/docker/overlay2/2e102ab7bfff01ee50fd08f077bb9897076e93e9b4d84390e1a8d1ffc40dd497/merged",
                "UpperDir": "/var/lib/docker/overlay2/2e102ab7bfff01ee50fd08f077bb9897076e93e9b4d84390e1a8d1ffc40dd497/diff",
                "WorkDir": "/var/lib/docker/overlay2/2e102ab7bfff01ee50fd08f077bb9897076e93e9b4d84390e1a8d1ffc40dd497/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "a2905ca0f71c",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "USER=herokuishuser",
                "DYNO=worker.1",
                "DOKKU_TRACE=",
                "CACHE_PATH=/cache",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DEBIAN_FRONTEND=noninteractive"
            ],
            "Cmd": [
                "/start",
                "worker"
            ],
            "Image": "dokku/earth:latest",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "com.dokku.app-name": "earth",
                "com.dokku.container-type": "deploy",
                "dokku": "",
                "dokku_phase_script": "predeploy",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.vendor": "dokku"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "833238ecb4b320dfcf83255e1ad49f2265d15133bf3fb2454fd9cfb5c0230ae9",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/833238ecb4b3",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "6a00b45f58efa71abf9825ebeae33d62e66a462d736fbb475a2625f73aff5de2",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]

$ docker logs a2905ca0f71c
(nothing)
```

**Comment by charleshan** [comment]
Your suggestion on trying `docker logs $CID` gave me some hints on how to fix the issue. I changed the logger level to see what's happening with the containing since the logs shouldn't be empty when it's exiting or getting killed. I fixed why the container was exiting as soon as it spawned and this solved the issue.

I still don't know how this is related to the rename error. I just wanted to report my findings. This was the reason why the worker containers were failing to stay alive: https://github.com/celery/celery/issues/4118

**Comment by zarqman** [comment]
Just ran into this. I was pretty baffled until finding this issue. just as here, it turned out that the problem was related to a container not starting correctly

Perhaps dokku could automatically display the results of docker logs for the failed-to-rename container and/or display a hint to check the logs and that the container is starting properly?

**Comment by arusa** [comment]
In my case the container was not starting because the partition with the docker volumes was full.

I found the reason using:

```
# find the container that can't start
docker ps -a
# show the logs of the container
docker logs <container_id>
```

**Comment by vanastassiou** [comment]
I was able to duplicate a similar problem by creating a low-RAM environment and deploying Celery apps whose workers repeatedly timed out on startup in two ways:
1. Throttle the RAM down enough that deployment of a new app takes ~20 minutes to complete (but does eventually complete)
1. Deploy with a bug that causes it to fail ~20% of the time, then run `ps:restart` on it until redeployment fails, which looks like this (see below cut)

<details><summary>Redeployment log</summary>

```
root@32e030c12c6a:/app# dokku ps:restart celery2
-----> Releasing celery2 (dokku/celery2:latest)...
-----> Deploying celery2 (dokku/celery2:latest)...
-----> App Procfile file found (/home/dokku/celery2/DOKKU_PROCFILE)
       DOKKU_SCALE declares scale -> web=1 worker=1
=====> Processing deployment checks
       No CHECKS file found. Simple container checks will be performed.
       For more efficient zero downtime deployments, create a CHECKS file. See http://dokku.viewdocs.io/dokku/deployment/zero-downtime-deploys/ for examples
-----> Attempting pre-flight checks (web.1)
       Waiting for 10 seconds ...
       Default container check successful!
-----> Attempting pre-flight checks (worker.1)
       Waiting for 10 seconds ...
baff804de98156e0841538f299a5851660d1bc16517b81a26f129197219aa1eb
App container failed to start!!
=====> celery2 worker container output:
       Traceback (most recent call last):
         File "/app/.heroku/python/bin/celery", line 11, in <module>
           sys.exit(main())
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/__main__.py", line 16, in main
           _main()
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/celery.py", line 322, in main
           cmd.execute_from_commandline(argv)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/celery.py", line 499, in execute_from_commandline
           super(CeleryCommand, self).execute_from_commandline(argv)))
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 289, in execute_from_commandline
           argv = self.setup_app_from_commandline(argv)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 509, in setup_app_from_commandline
           self.app = self.find_app(app)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 531, in find_app
           return find_app(app, symbol_by_name=self.symbol_by_name)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/app/utils.py", line 373, in find_app
           sym = symbol_by_name(app, imp=imp)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 534, in symbol_by_name
           return imports.symbol_by_name(name, imp=imp)
         File "/app/.heroku/python/lib/python3.6/site-packages/kombu/utils/imports.py", line 57, in symbol_by_name
           module = imp(module_name, package=package, **kwargs)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/utils/imports.py", line 111, in import_from_cwd
           return imp(module, package=package)
         File "/app/.heroku/python/lib/python3.6/importlib/__init__.py", line 126, in import_module
           return _bootstrap._gcd_import(name[level:], package, level)
         File "<frozen importlib._bootstrap>", line 994, in _gcd_import
         File "<frozen importlib._bootstrap>", line 971, in _find_and_load
         File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
         File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
         File "<frozen importlib._bootstrap_external>", line 678, in exec_module
         File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
         File "/app/index.py", line 10, in <module>
           from app import app, snap
         File "/app/app.py", line 9, in <module>
           n = 1 / 0    # fails 20% of the time
       ZeroDivisionError: division by zero
       Traceback (most recent call last):
         File "/app/.heroku/python/bin/celery", line 11, in <module>
           sys.exit(main())
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/__main__.py", line 16, in main
           _main()
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/celery.py", line 322, in main
           cmd.execute_from_commandline(argv)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/celery.py", line 499, in execute_from_commandline
           super(CeleryCommand, self).execute_from_commandline(argv)))
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 289, in execute_from_commandline
           argv = self.setup_app_from_commandline(argv)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 509, in setup_app_from_commandline
           self.app = self.find_app(app)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 531, in find_app
           return find_app(app, symbol_by_name=self.symbol_by_name)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/app/utils.py", line 373, in find_app
           sym = symbol_by_name(app, imp=imp)
         File "/app/.heroku/python/lib/python3.6/site-packages/celery/bin/base.py", line 534, in symbol_by_name
           return imports.symbol_by_name(name, imp=imp)
         File "/app/.heroku/python/lib/python3.6/site-packages/kombu/utils/imports.py", line 57, in symbol_by_name
           module = imp(module_name, package=package, **kwargs)
         File "/app/.herokutdown (MainProcess)
=====> end celery2 worker container output
```
</details>

I didn't get the `Error: failed to rename container named` error reported in the initial post, but the symptoms are similar enough that I think we're dealing with a similar issue. It looks like a successful worker start is necessary for Dokku to give the app containers their final "real" names vs the randomly-named build containers; as long as a worker is failing, the web container never gets its new name.

I'm looking at implementing a timeout somewhere in the deployment process that triggers a cleanup if *n* minutes pass without a worker finishing its startup, but I'm wondering if the Dokku maintenance team would have a tidier solution given their familiarity with the code.

**Comment by nileio** [comment]
Hi guys I am facing the same issue- the only thing I think it has to do with the restart-policy because the container doesn't start properly. This issue occurred for first time only after the host was restarted.

**Comment by josegonzalez** [comment]
Closing as there is a pull request open.
